### PR TITLE
Update ecosystem.rst to include Pint

### PR DESCRIPTION
**Pull request by znicholls**
We are working on upgrading pint to be compatible with pandas, see https://github.com/hgrecco/pint/pull/684

I am guessing that the line in the docs,

> If you’re building a library that implements the interface, please publicize it on Extension Data Types.

meant something like this pull request. If that's completely wrong, apologies.

- [x] closes #xxxx (N/A as not directly related to an issue, but makes progress towards #10349 )
- [x] (N/A) tests added / passed 
- [x] (N/A) passes `git diff upstream/master -u -- "*.py" | flake8 --diff` *
- [x] whatsnew entry

### CODE REVIEW

**Comment by jbrockmendel** [comment]
I’ll take a look at pint#684; units support have been an often-requested feature for pandas.

File: doc/source/ecosystem.rst
Comment (jreback) [inline doc/source/ecosystem.rst]: line should be the same length as title

File: doc/source/ecosystem.rst
Comment (jreback) [inline doc/source/ecosystem.rst]: can put a link here to pint docs

File: doc/source/whatsnew/v0.23.5.txt
Comment (jreback) [inline doc/source/whatsnew/v0.23.5.txt]: move to 0.24

**Review by jreback — CHANGES_REQUESTED** [review]
I assume that this needs a new release of pint yes? pls indicate the minimum version here.

File: doc/source/ecosystem.rst
Comment (jbrockmendel) [inline doc/source/ecosystem.rst]: This line seems out of place.  Is the idea that this PR goes on hold until the Pint PR is merged?

Comment (znicholls) [reply doc/source/ecosystem.rst]: yes sorry I should have changed to WIP earlier. 

@jreback's review asks for a minimum required Pint version. As I don't know what that version will be until the PR goes through, this PR also has to go on hold.

**Comment by znicholls** [comment]
We know the version number from https://github.com/hgrecco/pint/issues/693#issuecomment-419212581, can I do anything else?

**Comment by znicholls** [comment]
@jreback @jbrockmendel any idea why this is failing..?

**Comment by jbrockmendel** [comment]
> any idea why this is failing..?

Looks like a CI bug that was cleared up a couple days ago.  Try rebasing?

**Comment by znicholls** [comment]
We're working on a new release of Pint (https://github.com/hgrecco/pint/issues/693) but none of us can find the time right now, I hope to in January

**Comment by datapythonista** [comment]
CI is green now, is this ready to be merged, or are we waiting for the next Pint release?

**Comment by datapythonista** [comment]
@znicholls is this ready to be merged? or is a new Pint release or something else needed?

**Comment by znicholls** [comment]
> or is a new Pint release or something else needed?

Up to you guys. It will be included in the next Pint release but maybe best wait given we're still a bit uncertain about how best to do it (Panda's pytest test suite isn't playing nice with Pint's unittest test suite and we don't understand why, see https://github.com/hgrecco/pint/pull/741)

**Comment by datapythonista** [comment]
Ok, I see. Thanks for the clarification. I'll then close this PR for now, but please ping us when you update Pint, so we reopen it and discuss any needed change there.

**Comment by znicholls** [comment]
@datapythonista we have release pint-pandas on pypi (https://pypi.org/project/Pint-Pandas/). I've re-written the docs (see e.g. https://github.com/znicholls/pandas/blob/patch-1/doc/source/ecosystem.rst#pint-pandas) so can we re-open or make a new PR please? 

FYI @hgrecco and @andrewgsavage

**Comment by datapythonista** [comment]
GitHub doesn't let me reopen this PR, since the branch has been forced pushed or recreated. Can you open a new PR please?

Also, the ecosystem was moved to the website, it's a different file now.

Thanks for the work on this.

**Comment by znicholls** [comment]
ok cool have made #35170

**Pull request closed without merging**
